src/crimson/os/seastore/segment_manager/block.cc

```diff
@@ -557,6 +573,34 @@ BlockSegmentManager::mkfs_ret BlockSegmentManager::mkfs(
   });
 }
 
+BlockSegmentManager::mkfs_ret BlockSegmentManager::shard_mkfs(
+  device_config_t sm_config)
```

Comment:
I think the overall mkfs logic can be restructured with less repeating code, for example for each device:

`SeaStore::mkfs()` on the primary core only, call `Device::mkfs()`, mostly to (maybe) create the device and to write down the initial superblock.

`SeaStore::ShardStores::mkfs()` on each core (including the primary core), call `Device::mkfs_sharded()`, to include the common mkfs logic in each device shard.